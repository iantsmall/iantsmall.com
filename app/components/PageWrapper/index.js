/**
 *
 * PageWrapper
 *
 */

import React, { memo } from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Helmet } from 'react-helmet';
import Background from 'images/bg.jpg';
import ErrorBoundary from 'containers/ErrorBoundary';
import Menu from 'containers/Menu/Loadable';

const Wrapper = styled.div`
  width: 100vw;
  height: 100vh;
  margin: 0px;
  background-image: url(${Background});
  background-repeat: repeat-y;
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  overflow-x: hidden;
  @media (min-width: 100px) and (max-width: 728px) and (orientation: portrait) {
    margin: 0px;
    background: url(${Background}) fixed no;
    overflow-x: hidden;
  }
`;

const bootstrapCssLinkProps = {
  rel: 'stylesheet',
  href: 'https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css',
  integrity:
    'sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T',
  crossorigin: 'anonymous',
};

// const bootstrapCssLinkProps = {
// 	rel: "stylesheet",
// 	href: "https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css",
// 	integrity: "sha384-9gVQ4dYFwwWSjIDZnLEWnxCjeSWFphJiwGPXr1jddIhOegiu1FwO5qRGvFXOdJZ4",
// 	crossorigin: "anonymous",
// };

const PageWrapper = props => (
  <Wrapper>
    <Helmet>
      <link {...bootstrapCssLinkProps} />
    </Helmet>
    <ErrorBoundary {...props}>
      <Menu />
    </ErrorBoundary>
    <ErrorBoundary {...props} />
  </Wrapper>
);
PageWrapper.propTypes = {};

export default memo(PageWrapper);
