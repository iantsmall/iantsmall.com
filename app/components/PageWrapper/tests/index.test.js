/**
 *
 * Tests for PageWrapper
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
// import { render } from 'react-testing-library';
import render from 'utils/testRenderWithContext';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import PageWrapper from '../index';

describe('<PageWrapper />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <PageWrapper>
        <span>Some Test Text Content</span>
      </PageWrapper>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <PageWrapper>
        <div />
      </PageWrapper>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
