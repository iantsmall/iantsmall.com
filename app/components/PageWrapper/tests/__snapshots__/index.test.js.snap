// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWrapper /> Should render and match the snapshot 1`] = `
<div
  class="PageWrapper__Wrapper-az8zmc-0 dhvOGG"
>
  <div />
</div>
`;
